week2/H.work-week-6/script.js

```diff
@@ -0,0 +1,341 @@
+
+function doIt() {
+  console.log("I am done");
+}
+setTimeout(doIt, 2000);
+
+//2.1-We saw that we can pass functions as arguments to other functions. Your task is to write a function that takes another function as an argument and runs it.
+
+function foo(func) {
+  // What to do here?
+
+  console.log("Welcome ");
+  return func();
+}
+
+function bar() {
+  console.log("Hello, I am bar!");
+}
+
+foo(bar);
+
+/* 2.2 You must write a function that takes 4 arguments.
+
+A start value
+An end value
+A callback to call if the number is divisible by 3
+A callback to use if the number is divisible by 5
+The function should first generate an array containing values from start value to end value (inclusive).
+
+Then the function should take the newly created array and iterate over it, and calling the first callback if the array value is divisible by 3.
+
+The function should call the second callback if the array value is divisible by 5.
+
+Both functions should be called if the array value is divisible by both 3 and 5. */
+const myArray = [];
+let threePartition;
+let fivePartition;
+let threeFive;
+function threfive(startIndex, endIndex, threefunc, fivefunc) {
+  for (let i = startIndex; i <= endIndex; i++) {
+    myArray.push(i);
+  }
+  console.log(myArray);
+  threefunc = function() {
+    threePartition = myArray.filter(x => x % 3 === 0);
+  };
+
+  threefunc();
+  console.log(threePartition);
+
+  function fivefunc() {
+    fivePartition = myArray.filter(x => x % 5 === 0);
+  }
+  fivefunc();
+  console.log(fivePartition);
+
+  threeFive = myArray.filter(x => x % 15 === 0);
+  console.log(threeFive);
+}
+
+threfive(2, 33, threePartition, fivePartition);
+
+console.log(myArray);
+console.log(threePartition);
+console.log(fivePartition);
+console.log(threeFive);
+
+/*let a = [1, 2, 3, 4, 5];
+let b = a.filter(x => x % 2 === 0);
+console.log(b);  */
+
+//2.3 Please solve this problem:
+//Basic Algorithm Scripting: Repeat a String Repeat a String
+//2.3.1: with a for loop.
+//2.3.2: with a while loop.
+//2.3.3: with a do...while loop.
+
+//2.3.1: with a for loop.
+let bring = ""; //Burası çok önemli eğer bir değişkene herdöndüğünde toplamaya devam etsin ekleyerek gitsin dersek değişkeni tanımlayarak boş tırnaklar içinde tanımlayalım yoksa sadece tanımlarsak undefined şekilde ilk olarak undefined gelecektir
+let _repeat = function(x, y) {
+  for (let i = 1; i <= y; i++) {
+    y <= 0 ? bring : (bring += x);
+  }
+  return bring;
+};
+
+_repeat("Hello", 3);
+console.log(bring);
+
+//2.3.2: with a while loop.
+
+let _import = "";
+function repeat_(x, y) {
+  let i = 1;
+  while (i <= y) {
+    i++;
+    y <= 0 ? _import : (_import += x + " ");
+  }
+  return _import;
+}
+
+repeat_("Hi", 4);
+console.log(_import);
+
+//2.3.1: with a for loop.
+let results = "";
+let getir = function(x, y) {
```

Comment:
Kodunuzun daha okunakli olmasi adina degisken isimlerinin uygun verilmesi genelde tavsiye edilir

```suggestion
let getir = function(str, repeat) {
```